src/plugins/formulas/__tests__/initialization.spec.js

```diff
@@ -0,0 +1,699 @@
+
+          expect(plugin.sheetId).toEqual(hfInstance1.getSheetId(plugin.sheetName));
+
+          expect(getData()).toEqual(hfInstance1.getSheetSerialized(plugin.sheetId));
+          expect(hfInstance1.doesSheetExist(plugin.sheetName)).toEqual(true);
+          expect(hfInstance1.getSheetSerialized(plugin.sheetId)).toEqual([]);
+        });
+      });
+    });
+  });
+
+  describe('Updating HF settings', () => {
+    it('should initialize HyperFormula with the default set of settings', () => {
+      handsontable({
+        formulas: {
+          hyperformula: HyperFormula
+        },
+        licenseKey: 'non-commercial-and-evaluation'
+      });
+
+      const plugin = getPlugin('formulas');
+      const hfConfig = plugin.hyperformula.getConfig();
+
+      expect(hfConfig.binarySearchThreshold).toEqual(20);
+      expect(hfConfig.matrixDetection).toEqual(true);
+      expect(hfConfig.matrixDetectionThreshold).toEqual(100);
+      expect(hfConfig.useColumnIndex).toEqual(false);
+      expect(hfConfig.useStats).toEqual(false);
+      expect(hfConfig.evaluateNullToZero).toEqual(true);
+      expect(hfConfig.precisionEpsilon).toEqual(1e-13);
+      expect(hfConfig.precisionRounding).toEqual(14);
+      expect(hfConfig.smartRounding).toEqual(true);
+      expect(hfConfig.leapYear1900).toEqual(true);
+    });
+
+    it('should update the HyperFormula config with the default set of settings', () => {
+      const hfInstance1 = HyperFormula.buildEmpty({ licenseKey: 'internal-use-in-handsontable' });
+      hfInstance1.updateConfig({
+        binarySearchThreshold: 25,
+        matrixDetection: false,
+        matrixDetectionThreshold: 125,
+        useColumnIndex: true,
+        useStats: true,
+      });
+
+      handsontable({
+        formulas: {
+          hyperformula: hfInstance1
+        },
+        licenseKey: 'non-commercial-and-evaluation'
+      });
+
+      const plugin = getPlugin('formulas');
+      const hfConfig = plugin.hyperformula.getConfig();
+
+      expect(hfConfig.binarySearchThreshold).toEqual(20);
+      expect(hfConfig.matrixDetection).toEqual(true);
+      expect(hfConfig.matrixDetectionThreshold).toEqual(100);
+      expect(hfConfig.useColumnIndex).toEqual(false);
+      expect(hfConfig.useStats).toEqual(false);
+      expect(hfConfig.evaluateNullToZero).toEqual(true);
+      expect(hfConfig.precisionEpsilon).toEqual(1e-13);
+      expect(hfConfig.precisionRounding).toEqual(14);
+      expect(hfConfig.smartRounding).toEqual(true);
+      expect(hfConfig.leapYear1900).toEqual(true);
+    });
+
+    it('should update the HyperFormula configuration with the options defined in `formulas.hyperFormulaConfig`', () => {
+      handsontable({
+        formulas: {
+          hyperformula: HyperFormula,
+          hyperformulaConfig: {
+            binarySearchThreshold: 25,
+            matrixDetection: false,
+            matrixDetectionThreshold: 125,
+            useColumnIndex: true,
+            useStats: true,
+          }
```

Comment:
When you give an instance and config you may expect that every Handsontable instance will apply a new config settings for HF. Or it may be confusing if you need to give the same settings, or different, or ignore the setting.

I wanted to avoid this by having a single config key for either instance or config.

Reply:
That makes sense. So passing a config object would be equivalent to passing a `HyperFormula` class + adding config? (So, basically, creating a fresh HF instance internally?)

Reply:
We still need the HF class but if it is a part of the config you can set either config with class or an instance (that already is configured). Never both